packages/wonder-blocks-search-field/src/components/search-field.tsx

```diff
@@ -146,13 +174,14 @@ const SearchField: React.ForwardRefExoticComponent<
 
     // @ts-expect-error [FEI-5019] - TS2322 - Type '() => JSX.Element | null' is not assignable to type '() => ReactElement<any, string | JSXElementConstructor<any>>'.
     const maybeRenderClearIconButton: () => React.ReactElement = () => {
-        if (!value.length) {
+        if (!value.length || disabled) {
             return null;
         }
 
         return (
             <IconButton
                 icon={xIcon}
+                size="xsmall"
```

Comment:
I also updated the dismiss icon styling (bold icon, smaller, spacing!). Thanks for catching this @jandrade! 

Since IconButton is smaller now, this addresses the issue where the focus outline on the dismiss button can be cropped out if an ancestor element has `overflow: hidden` (see [Chromatic comment](https://www.chromatic.com/test?appId=5e1bf4b385e3fb0020b7073c&id=673baf9294168df34de5619b))
![image](https://github.com/user-attachments/assets/51ddd0a8-fe6a-40f2-9620-260b6dfe9f52)

The hit area seems too small though for accessibility. I wonder if all sizes of the IconButton component should all have the same minimum hit area?

Reply:
I think we should probably use `size="small"` instead, as it keeps the same icon size (16x16) but the hit area is slightly bigger (32 instead of 24), which would improve a11y in that regard. We'd still fail to achieve AAA, but I think it would be a good compromise for the mid-term.

And I like the suggestion about revisiting the hit areas of all `IconButton` variants. I think this is something worth exploring and documenting in the WB audit that we are doing for the new experience.

Reply:
`size="small"` for IconButton uses a [24x24 icon](https://khan.github.io/wonder-blocks/?path=/docs/packages-iconbutton--docs#sizes), which is different from `size="small"` for PhosphorIcon! I'll update this to `size="small"` and have the icon be a bit bigger so it is more usable!

Reply:
I'll also update it back to the regular icon weight, I think when it's bold and 24x24, the dismiss icon is too prominent! 
![image](https://github.com/user-attachments/assets/39dc6139-fdd1-4f46-98d5-80cfe92310e0) ![image](https://github.com/user-attachments/assets/34366fca-8993-448b-bc3d-c8adc9e27154)

Reply:
Thanks for trying this out. The last iteration makes sense to me.

For context, I got a big confused as I saw that the designs apparently used an icon of 24x24 but in reality the icon was 16x16

<img width="630" alt="Screenshot 2024-11-20 at 2 48 50 PM" src="https://github.com/user-attachments/assets/9046256b-c1ff-4d1b-a092-fc52c7625b88">

https://www.figma.com/design/VbVu3h2BpBhH80niq101MHHE/%F0%9F%92%A0-Main-Components?node-id=13843-9211&t=nNY8Jl9Ah253UTqj-4

Reply:
Ahh yes that is confusing! I added a note in WB-1792 to document tips for developers around icons once we work with design to make the translation easier!

I also updated the [audit spreadsheet](https://docs.google.com/spreadsheets/d/1Wcnv-JKnyT-w2MG-dAZsS5PacV8IoUqYbzNYQq-mxJY/edit?gid=0#gid=0) to include notes about the IconButton hit area and IconButton/PhosphorIcon size consistency!